PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -5798,9 +5788,9 @@ function Show-SddcDiagnosticReport
     # Extract ZIP if neccesary
     $Path = Check-ExtractZip $Path
 
-    # Repror Output file
-    $ReportFileName = (Convert-Path .) + "\" + $Path.Trim('\').Split('\')[-1] + "-SDDCDiagnosticReport.txt"
-    Start-Transcript -Path $ReportFileName -NoClobber
+    if(($OutputFile) -and (Test-Path -IsValid $OutputFile)){

```

Comment:
I don't think -IsValid does what you think it does here. Its just checking for whether a specified drive letter is valid, and beyond that only syntactic plausibility that the remainder could be a path. Its not validating presence of any component of the path in the filesystem. Start-Transcript will do all that.

Should we take a -NoClobber defaulting to $true that a caller could then specify as -NoClobber:$false to allow overwrite? You can call Start-Transcript -NoClobber:$somevalue